source/entities/GameEntity.cpp

```diff
@@ -616,4 +615,20 @@ void GameEntity::notifyFightPlayer(Tile* tile)
     getGameMap()->playerIsFighting(getSeat()->getPlayer(), tile);
 }
 
+void GameEntity::setParentNodeDetach(uint32_t mask, bool value)
+{
+    // We save the current attach  state
+    bool oldState = (mEntityParentNodeAttach == EntityParentNodeAttach::ATTACHED);
+    // We compute the new attach state
+    mEntityParentNodeAttach = (value ? mEntityParentNodeAttach | mask : mEntityParentNodeAttach & ~mask);
+
+    // If the attach state changed, we do what is needed
+    bool newState = (mEntityParentNodeAttach == EntityParentNodeAttach::ATTACHED);
+    if(oldState == newState)
+        return;
```

Comment:
if(oldState != newState){
    if(newState)
        RenderManager::getSingleton().rrAttachEntity(this);
    else
        RenderManager::getSingleton().rrDetachEntity(this);
}